pkg/controllers/managementuser/secret/project_scoped_secrets.go

```diff
@@ -81,28 +103,49 @@ func (n *namespaceHandler) OnChange(_ string, namespace *corev1.Namespace) (*cor
 	return namespace, n.removeUndesiredProjectScopedSecrets(namespace, desiredSecrets)
 }
 
-// getProjectScopedSecretsFromNamespace gets all project scoped secret from a project namespace.
-func (n *namespaceHandler) getProjectScopedSecretsFromNamespace(namespace *corev1.Namespace) ([]*corev1.Secret, error) {
-	// check if namespace is part of a project
-	projectID, ok := namespace.Annotations[projectIDLabel]
-	if !ok {
-		return nil, nil
-	}
+// migrateExistingProjectScopedSecrets migrates existing project scoped secrets.
+// It removes the following:
+//   - Finalizer "clusterscoped.controller.cattle.io/secretsController_<clusterID>"
+//   - Annotation "lifecycle.cattle.io/create.secretsController_<clusterID>"
+//   - Label "rancher.io/creator"
+//
+// And adds:
+//   - Label "management.cattle.io/project-scoped-secret"
+func (n *namespaceHandler) migrateExistingProjectScopedSecrets(project *v3.Project) error {
+	backingNamespace := project.GetProjectBackingNamespace()
+	clusterName := project.Spec.ClusterName
 
-	clusterName, projectName, found := strings.Cut(projectID, ":")
-	if !found {
-		logrus.Debugf("Namespace %s projectId annotation %s is malformed, should be <cluster name>:<project name>", namespace.Name, namespace.Annotations[projectIDLabel])
-		return nil, nil
+	r, err := labels.NewRequirement(normanLabel, selection.Equals, []string{"norman"})
+	if err != nil {
+		return err
 	}
-
-	p, err := n.projectCache.Get(clusterName, projectName)
+	secrets, err := n.managementSecretCache.List(backingNamespace, labels.NewSelector().Add(*r))
 	if err != nil {
-		return nil, fmt.Errorf("error getting project %s for namespace %s: %w", projectName, namespace.Name, err)
+		return err
 	}
 
-	backingNamespace := p.GetProjectBackingNamespace()
+	// Remove the norman creator label, secret lifecycle annotation and secret lifecycle finalizer.
+	// The controllers handling those were removed in v2.12 https://github.com/rancher/rancher/pull/49995
+	var errs error
+	for _, s := range secrets {
+		delete(s.Labels, normanLabel)
```

Comment:
IMHO we have to `DeepCopy()` before making changes to avoid modifications in the cache.